### ISSUE TITLE
Reuse AWS instances

### ISSUE THREAD

**Issue by PGijsbers** [openml/automlbenchmark#171]
This is a suggestion I found in the TODO:

We can reuse an AWS instance instead of shutting them down after each job.
At least during a single benchmark, we could limit #instances = #parallel jobs.

I am not sure how this would actually work, in principle I see there is a speed up:
 - only have to create the instance once
 - only have to download a dataset once (if subsequent jobs use the same dataset)
 - only have to install the automl framework once (if subsequent jobs use the same framework)

I also see some pitfalls:
 - When re-using an instance for multiple frameworks, the second framework might be affected by leftovers of the installation of the first.
 - To provide the same disk space, does this mean we have to make sure to transfer/delete all (cache) files of the previous run?
 - Does this require additional communication? The most naive way I can imagine of doing this is to just run all folds of a (framework, task) tuple per instance, which should not require additional communication (though some extra clean up, see above). This wouldn't allow perfect re-use but it's probably pretty good.
 - (for the future:) Is there an increased risk of interruption when using spot instances? Is the effect greater? I assume the answer is no to both (interruption is just based on bid-price, and when transferring results between each part of the job (e.g. fold), no more than one is lost), but I am not sure.

**Comment by sebhrusen** [comment]
@PGijsbers yes, this is an old suggestion, and I agree with your pitfalls, so I would not consider this as a priority.
Good to keep this for reference though.

**Comment by PGijsbers** [comment]
Removing TODOs from code (#172), so leaving it here:

https://github.com/openml/automlbenchmark/blob/98bf554dc07fc1cd90a254b4cb852986b483dd94/amlb/runners/aws.py#L802-L804
